### PR TITLE
[Breaking Change]: Update react-query to v5.62.0

### PR DESCRIPTION
**Pull request by karan-leapwallet**
## Description

This PR updates the react-query version to 5.62.0 and removes QueryClientProvider initialisation from Graz Provider. As a results, dApps must now wrap Graz provider with QueryClientProvider on their end. Also note that react-query has been added as peer dependency now. All the examples have been updated to reflect the same.

## Checklist

- [x] I have made sure the upstream branch for this PR is correct
- [x] I have made sure this PR is ready to merge
- [x] I have made sure that I have assigned reviewers related to this project

## Changes

- [x] Changed react-query version 5.62.0
- [x] Removed QueryClientProvider initialisation from GrazProvider

## Screenshots

...

## Testing

- Open page ...
- Click ...
- Make sure that ...

## Links/References

- ...
- ...
- ...

## Notes

- ...
- ...
- ...

### CODE REVIEW

File: README.md
Comment (siddharthborderwala) [inline README.md:1]: Need to show `QueryClientProvider` import and `queryClient` instantiation.

File: docs/docs/index.md
Comment (siddharthborderwala) [inline docs/docs/index.md:1]: Need to show `QueryClientProvider` import and `queryClient` instantiation.

File: docs/docs/provider/grazProvider.md
Comment (siddharthborderwala) [inline docs/docs/provider/grazProvider.md:1]: Need to show `QueryClientProvider` import and `queryClient` instantiation.

**Review by siddharthborderwala — APPROVED** [review]

File: README.md
Comment (karan-leapwallet) [reply README.md:1]: Updated

File: docs/docs/index.md
Comment (karan-leapwallet) [reply docs/docs/index.md:1]: Updated

File: docs/docs/provider/grazProvider.md
Comment (karan-leapwallet) [reply docs/docs/provider/grazProvider.md:1]: Updated

**Comment by codingki** [comment]
great PR, looking forward!

**Review by codingki — APPROVED** [review]
LGTM!

**Pull request merged**